tools/src/main/scala/akka/contrib/persistence/mongodb/MigrateToSuffixedCollections.scala

```diff
@@ -0,0 +1,206 @@
+/* 
+ * Contributions:
+ * Jean-Francois GUENA: implement "suffixed collection name" migration tool
+ * ...
+ */
+package akka.contrib.persistence.mongodb
+
+import akka.actor.ActorSystem
+import com.mongodb.casbah.Imports._
+import com.typesafe.config.Config
+
+import scala.util.Try
+
+class MigrateToSuffixedCollections(system: ActorSystem, config: Config) extends CasbahMongoDriver(system, config) {
+
+  def migrateToSuffixCollections(): Unit = {
+    import akka.contrib.persistence.mongodb.JournallingFieldNames._
+    import scala.util.Random
+
+    // INIT //
+
+    logger.info("Starting automatic migration to collections with suffixed names\nThis may take a while...")
+
+    if (settings.JournalAutomaticUpgrade) {
+      logger.warn("Please, disable 'journal-automatic-upgrade' option when migrating from unique to suffixed collections. Aborting...")
+      return
+    }
+
+    if (!useSuffixedCollectionNames) {
+      logger.warn("Please, provide some 'suffix-builder.class' option when migrating from unique to suffixed collections. Aborting...")
+      return
+    }
+
+    // create a temporary collection
+    val temporaryCollectionName = s"migration2suffix-${System.currentTimeMillis()}-${Random.nextInt(1000)}"
+    val temporaryCollection = collection(temporaryCollectionName)
+
+    def pidQuery(persistenceId: String) = (PROCESSOR_ID $eq persistenceId)
+
+    // JOURNALS //
+
+    // aggregate persistenceIds, contained in unique journal, and store them in temporary collection
+    journal.aggregate(
+      MongoDBObject("$group" -> MongoDBObject("_id" -> s"$$$PROCESSOR_ID")) ::
+        MongoDBObject("$out" -> temporaryCollectionName) ::
+        Nil).results
+
+    // for each retrieved persistenceId, insert corresponding records from unique journal
+    // to appropriate (and newly created) suffixed journal, then remove them from unique journal
+    if (temporaryCollection.count() > 0) {
```

Comment:
This block is complicated - can it be broken into methods that are chained together?